docs/src/components/SelectPropControls.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import { Flex, Select, SelectProps } from '@aws-amplify/ui-react';
+
+import { FieldLabeler } from './FieldLabeler';
+
+export interface SelectPropContorlsProps extends SelectProps {
+  setSize: (value: React.SetStateAction<SelectProps['size']>) => void;
+  setVariation: (value: React.SetStateAction<SelectProps['variation']>) => void;
+  setIsDisabled: (
+    value: React.SetStateAction<SelectProps['isDisabled']>
+  ) => void;
+}
+
+interface SelectPropControlsInterface {
+  (props: SelectPropContorlsProps): JSX.Element;
+}
+
+export const SelectPropControls: SelectPropControlsInterface = ({
+  setSize,
+  setVariation,
+  setIsDisabled,
+}) => {
+  return (
+    <fieldset className="p-4 border-2 border-current border-solid">
+      <legend className="font-bold p-1">Select props:</legend>
+      <Flex justifyContent="space-between">
+        <FieldLabeler id="size">
+          <Select
+            name="size"
+            id="size"
+            defaultValue=""
+            onChange={(event) =>
+              setSize(event.target.value as SelectProps['size'])
+            }
+            width="8rem"
```

Comment:
It wouldn't. I also try `width="fit-content'`, looks like we have to specify a decent width explicitly :( 

This is because we need to respect the icon to the right, otherwise it will overlap with the content.

Reply:
Hmm, this sounds like a problem a customer would run into that we should fix at the Select component? Could we use extra `padding-inline-end` to prevent the option text from going underneath the icon?

Reply:
Humm, i think `padding-inline-end` will have no difference with `padding`? To fully address it, I think we should use percentage for `padding` so that the padding for field could be adjusted according to the `width`. And the icon could use the same percentage for its `right`

Reply:
Okay, it works. It will override the `padding-right` for select.